quixstreams/dataframe/dataframe.py

```diff
@@ -149,8 +159,24 @@ def stream(self) -> Stream:
         return self._stream
 
     @property
-    def topic(self) -> Topic:
-        return self._topic
+    def stream_id(self) -> str:
+        """
+        An identifier of the data stream this StreamingDataFrame
+        manipulates in the application.
+
+        It is used as a common prefix for state stores and group-by topics.
+        A new `stream_id` is set when StreamingDataFrames are merged via `.merge()`
+        or grouped via `.group_by()`.
+
+        StreamingDataFrames with different `stream_id` cannot access the same state stores.
+
+        By default, a topic name or a combination of topic names are used as `stream_id`.
+        """
+        return self._topic_manager.stream_id_from_topics(*self.topics)
```

Comment:
Followup on a comment on the `stream_id_from_topics` method signature.
```suggestion
        return self._topic_manager.stream_id_from_topics(self._topics)
```